### PR TITLE
v4.1.x/UCX/PML/SPML/OSC: fix compiler warnings

### PR DESCRIPTION
**Pull request by karasevb**
Fixes compiler warnings for v4.1.x and cherry-picked https://github.com/open-mpi/ompi/pull/9997, https://github.com/open-mpi/ompi/pull/9438

bot:notacherrypick

### CODE REVIEW

**Comment by karasevb** [comment]
bot:notacherrypick

**Comment by karasevb** [comment]
@jsquyres I don't have permission or anything to restart GitHub Action CI, the button "Re-run jobs" doesn't appear for me. Do you have any ideas?

**Comment by jsquyres** [comment]
You need to put `bot:notacherrypick` in the PR description, not a comment.

File: oshmem/mca/spml/ucx/spml_ucx.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ void mca_spml_ucx_peer_mkey_cache_init(mca_spml_ucx_ctx_t *ucx_ctx, int pe)
 int mca_spml_ucx_peer_mkey_cache_add(ucp_peer_t *ucp_peer, int index)
 {
     /* Allocate an array to hold the pointers to the ucx_cached_mkey */
-    if (index >= ucp_peer->mkeys_cnt){
+    if (index >= (int)ucp_peer->mkeys_cnt){
```
Comment (brminich) [inline oshmem/mca/spml/ucx/spml_ucx.c:124]: minor: space before `{`

**Review by brminich — APPROVED** [review]

File: oshmem/mca/spml/ucx/spml_ucx.c (hunk 1 above)
Comment (karasevb) [reply oshmem/mca/spml/ucx/spml_ucx.c]: fixed

**Pull request merged**